File: Dante1/Module1.06/Zadanie6.12/main.c
#include <stdio.h>

int main()
{
    int matrix[10][10];
    printf("Enter:\n");

    for(int i = 0; i < 10; i++)
    {
        for(int j = 0; j < 10; j++)
        {
            int check = scanf("%i", &matrix[i][j]);
            if(check != 1)
            {
                printf("\nIncorrect input");
                return 1;
            }
        }
    }

    int result = 0;
    for(int i = 1; i < 9; i++)
    {
        for(int j = 1; j < 9; j++)
        {
            int counter = 0;
            for(int i1 = -1; i1 < 2; i1++)
            {
                for(int j1 = -1; j1 < 2; j1++)
                {
                    if(i1 == 0 && j1 == 0)
                    {
                        continue;
                    }
                    if(matrix[i][j] < matrix[i + i1][j + j1])
                    {
                        counter++;
                    }
                }
            }
            if(counter == 8)
            {
                result++;
            }
        }
    }

    if(result == 0)
    {
        printf("\nNothing to show\n");
    }
    else
    {
        printf("%i\n", result);
    }

    for(int i = 1; i < 9; i++)
    {
        for(int j = 1; j < 9; j++)
        {
            int counter = 0;
            for(int i1 = -1; i1 < 2; i1++)
            {
                for(int j1 = -1; j1 < 2; j1++)
                {
                    if(i1 == 0 && j1 == 0)
                    {
                        continue;
                    }
                    if(matrix[i][j] < matrix[i + i1][j + j1])
                    {
                        counter++;
                    }
                }
            }
            if(counter == 8)
            {
                printf("%i %i\n", i, j);
            }
        }
    }

    return 0;
}

File: Dante1/Module1.06/Zadanie6.12/main.rs
fn main()
{
    let mut array:[[i32; 10]; 10] = [[0; 10]; 10];
    let mut input: String = String::new();

    println!("Enter: ");

    for i in 0..10
    {
        for j in 0..10
        {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Incorrect input");
            array[i][j] = input.trim().parse().expect("Incorrect number");
        }
    }

    let mut result: i32 = 0;

    for i in 1..9
    {
        for j in 1..9
        {
            let mut counter: i32 = 0;

            for a in -1..2
            {
                for b in -1..2
                {
                    if a == 0 && b == 0
                    {
                        continue;
                    }

                    if array[i][j] < array[i + a][j + b]
                    {
                        counter += 1;
                    }
                }
            }

            if counter == 8
            {
                result += 1;
            }
        }
    }
    
    if result == 0
    {
        panic!("Nothing to show");
    }
    else
    {
        println!("{}", result);
    }

    for i in 1..9
    {
        for j in 1..9
        {
            let mut counter: i32 = 0;

            for a in -1..2
            {
                for b in -1..2
                {
                    if a == 0 && b == 0
                    {
                        continue;
                    }

                    if array[i][j] < array[i + a][j + b]
                    {
                        counter += 1;
                    }
                }
            }

            if counter == 8
            {
                println!("{} {}", i, j);
            }
        }
    }
}
